Locks.cpp

```diff
@@ -0,0 +1,59 @@
+#include<iostream>
+#include<deque>
+#include<string>
+#include"Process.hpp"
+#include"Undefined.hpp"
+#include"Locks.hpp"
+void Lock::Show_List_of_Processes()
+{
+	std::cout << "Names of processes under lock: " << std::endl;
+	int i = 1;
+	for (auto process : waiting_process_list)
+	{
+		std::cout << i << ". " << process.get().getName() << std::endl;
+		i++;
+	}
+}
+std::string Lock::GetProcessName()

```

Comment:
Zdecyduj się, czy metoda ta ma na celu wypisanie nazwy pierwszego procesu pod zamkiem czy zwrócenie jej - nie na obie rzeczy naraz (bo jeśli ktoś będzie potrzebował drugiej funkcjonalności, to wątpliwe, by było przez niego pożądane wypisanie czegokolwiek naraz z pobraniem).

Sugerowałbym rozdzielenie na dwie funkcje, get oraz display.